weddineo-core/src/main/java/com/weddineo/configuration/filter/FirebaseFilter.java

```diff
@@ -0,0 +1,75 @@
+package com.weddineo.configuration.filter;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseAuthException;
+import com.google.firebase.auth.FirebaseToken;
+import com.weddineo.firebase.exception.WeddineoFirebaseRuntimeException;
+
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import lombok.Builder;
+
+@Builder
+public class FirebaseFilter extends OncePerRequestFilter {
```

Comment:
You should write some unit tests of that class - it implements a critical application logic.

Reply:
will be in the future